x-pack/plugins/ui_actions_enhanced/server/plugin.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { identity } from 'lodash';
+import { CoreSetup, Plugin, SavedObjectReference } from '../../../../src/core/server';
+import { EmbeddableSetup } from '../../../../src/plugins/embeddable/server';
+import { dynamicActionEnhancement } from './dynamic_action_enhancement';
+import {
+  ActionFactoryRegistry,
+  SerializedEvent,
+  ActionFactoryDefinition,
+  DynamicActionsState,
+} from './types';
+import { PersistableState } from '../../../../src/plugins/kibana_utils/common';
+
+export interface SetupContract {
+  registerActionFactory: any;
+}
+
+export type StartContract = PersistableState<DynamicActionsState>;
+
+interface SetupDependencies {
+  embeddable: EmbeddableSetup; // Embeddable are needed because they register basic triggers/actions.
+}
+
+export class AdvancedUiActionsPublicPlugin
+  implements Plugin<SetupContract, StartContract, SetupDependencies> {
+  protected readonly actionFactories: ActionFactoryRegistry = new Map();
+
+  constructor() {}
+
+  public setup(core: CoreSetup, { embeddable }: SetupDependencies) {
+    // this.enhancements = new UiActionsServiceEnhancements({
+    //   getLicense: () => this.getLicenseInfo(),
+    //   featureUsageSetup: licensing.featureUsage,
+    //   getFeatureUsageStart: () => startServices().plugins.licensing.featureUsage,
+    // });
+    embeddable.registerEnhancement(dynamicActionEnhancement(this));
+
+    return {
+      registerActionFactory: this.registerActionFactory,
```

Comment:
Shouldn't drilldown integrate with this? Looks like this is not used on the server

Reply:
on the client we have a helper function to register a drilldown, which in turn actionly registers action factory. on the server we have no need for that as registering a drilldown looks exactly the same as registering normal action factory due to the fact that registries on the server contain just the state management functions.

Reply:
but don't we need the drilldown specific functions for extracting/injecting the drilldown references?

Reply:
we will, but they are not part of this PR